internal/providers/terraform/azure/container_registry.go

```diff
@@ -0,0 +1,181 @@
+package azure
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetAzureRMContainerRegistryRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_container_registry",
+		RFunc: NewAzureRMContainerRegistry,
+	}
+}
+
+func NewAzureRMContainerRegistry(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	var nambersOflocations int
+	var storageGb, includedStorage, monthlyBuildVcpu *decimal.Decimal
+	var overStorage decimal.Decimal
+
+	sku := "Classic"
+	location := d.Get("location").String()
+
+	if d.Get("sku").Exists() {
+		sku = d.Get("sku").String()
+	}
+
+	switch sku {
+	case "Classic":
+		includedStorage = decimalPtr(decimal.NewFromFloat(10))
+	case "Basic":
+		includedStorage = decimalPtr(decimal.NewFromFloat(10))
+	case "Standard":
+		includedStorage = decimalPtr(decimal.NewFromFloat(100))
+	case "Premium":
+		includedStorage = decimalPtr(decimal.NewFromFloat(500))
+	}
+
+	if d.Get("georeplication_locations").Exists() {
+		nambersOflocations = len(d.Get("georeplication_locations").Array())
+	}
+	if d.Get("georeplications").Exists() {
+		nambersOflocations = len(d.Get("georeplications.0.location").Array())
+	}
+
+	costComponents := make([]*schema.CostComponent, 0)
+
+	if d.Get("georeplication_locations").Exists() || d.Get("georeplications").Exists() {
```

Comment:
should this be like `d.Get("value_name").Type != gjson.Null`? basically anytime I see `Exists()`, I think of this from our contributing doc:
In the past we used `.Exists()` but this only checks that the key does not exist in the Terraform JSON, not that the key exists and is set to null.